torchvision/csrc/io/decoder/subtitle_stream.cpp

```diff
@@ -44,20 +44,28 @@ int SubtitleStream::analyzePacket(const AVPacket* packet, bool* gotFrame) {
   // clean-up
   releaseSubtitle();
   // check flush packet
-  AVPacket avPacket;
-  av_init_packet(&avPacket);
-  avPacket.data = nullptr;
-  avPacket.size = 0;
-  auto pkt = packet ? *packet : avPacket;
+  AVPacket* avPacket;
+  avPacket = av_packet_alloc();
+  if (avPacket == NULL) {
+    LOG(ERROR) << "decoder as not able to allocate the packet.";
```

Comment:
See Joao's comment on the `nullptr`. Also, should we be returning an error if allocating a packet fails? Finally it's unclear to me why we allocate a packet even when `packet` is not null. This can be improved on a future PR.

Reply:
small enough change that I can push it now I take it, and revert if it breaks something profoundly